### PR TITLE
Add pytest to dev dependencies

### PR DESCRIPTION
**Pull request by virenbajaj**
Add `pytest` version 8.3.3 (latest) to the `dev` group of dependencies in poetry. 

Note: If development and test dependencies increase and diverge, we could create a new dependency group for testing.  

Reference: https://python-poetry.org/docs/managing-dependencies/

### CODE REVIEW

**Comment by dylanbouchard** [comment]
Thanks Viren! Can you please also include `pytest-asyncio`?

**Review by dylanbouchard — CHANGES_REQUESTED** [review]
Thanks Viren! Can you please also include `pytest-asyncio`?

File: pyproject.toml [hunk 1]
```diff
@@ -42,11 +42,13 @@ asyncio = "^3.4.3"
 sentence-transformers = "^2.7.0"
 typing = "<3.10"
 tiktoken = "^0.7.0"
+pytest-asyncio = "^0.24.0"
```
Comment (dylanbouchard) [inline pyproject.toml:45]: This should be under dev dependencies

**Review by dylanbouchard — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (virenbajaj) [reply pyproject.toml]: Thanks for catching it - fixed

**Pull request merged**